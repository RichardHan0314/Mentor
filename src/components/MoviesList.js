import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MovieCard from './MovieCard';

const MoviesList = ({ movies }) => {
    return (
        <>
            <MoviesCount>
                <b>{movies.length}</b> movies found
            </MoviesCount>
            <Wrapper>
                {movies.map((movie) => (
                    <Link to={`/movie/${movie.id}`} key = {movie.id}>
                        <MovieCard
                            movie = {movie}
                            key = {movie.id}
                            image = {movie.poster_path}
                            title = {movie.title}
                            genres = {movie.genres}
                            date = {movie.release_date}
                        />
                    </Link>
                   
                ))}
            </Wrapper>
        </>
    );
};

const Wrapper = styled.section`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    a {
        width: 30%;
    }
`;

const MoviesCount = styled.p`
    color: #fff;
    margin: 0;
    padding: 20px 0 10px;
`;

MoviesList.propTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    movies: PropTypes.array.isRequired,
};

export default MoviesList;
