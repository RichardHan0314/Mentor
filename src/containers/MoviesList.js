import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MovieCard from '../components/MovieCard';

const MoviesList = (props) => {
    const { movies, sortValue, deleteMovie, editMovie } = props;
    const moviesList = [...movies];

    moviesList.sort((a,b) => {
        return a[sortValue] > b[sortValue] ? 1 : -1;
    });
    
    return (
        <>
            <MoviesCount>
                <b>{moviesList.length}</b> movies found
            </MoviesCount>
            <Wrapper>
                {moviesList.map((movie) => (
                    <MovieCard
                        movie = {movie}
                        image = {movie.image}
                        title = {movie.title}
                        genre = {movie.genre}
                        date = {movie.date}
                        key = {movie.id}
                        deleteMovie = {deleteMovie}
                        editMovie = {editMovie}
                    />
                ))}
            </Wrapper>
        </>
    );
};

const Wrapper = styled.section`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
`;

const MoviesCount = styled.p`
    color: #fff;
    margin: 0;
    padding: 20px 0 10px;
`;

MoviesList.propTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    movies: PropTypes.array.isRequired,
    sortValue: PropTypes.string.isRequired,
    deleteMovie: PropTypes.func,
    editMovie: PropTypes.func
};

MoviesList.defaultProps = {
    deleteMovie: () => {},
    editMovie: () => {}
};

export default MoviesList;
