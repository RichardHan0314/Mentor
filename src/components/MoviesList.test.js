import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';
import MoviesList from './MoviesList';

describe('MoviesList component', () => {
    const defaultMovies = [];
    const moviesDataMock = [
        {
            id: 1,
            image: 'https://images-na.ssl-images-amazon.com/images/I/71niXI3lxlL._AC_SY679_.jpg',
            title: 'Avengers',
            genre: 'Action',
            date: '2016',
            url: '',
            overview: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam accumsan commodo ligula, non gravida arcu imperdiet sed. Donec eu nibh id lectus molestie volutpat non vel sapien. Proin nec imperdiet lectus.',
            runtime: '146',
            rating: '4.8'
        }
    ];

    it('Render NotFindURL snapshot', () => {

        const component = shallow(
            <MemoryRouter>
                <MoviesList movies={defaultMovies} />
            </MemoryRouter>
        );
        expect(component).toMatchSnapshot();
    });

    it('should be not have link with movie id', () => {
        const component = shallow(
            <MemoryRouter>
                <MoviesList movies={defaultMovies}>
                    <Link to="/movie/" />
                </MoviesList>
            </MemoryRouter>
        );
        expect(component.find(Link).props().to).toBe('/movie/');
    });

    it('should be have link with movie id', () => {
        const component = shallow(
            <MemoryRouter>
                <MoviesList movies={moviesDataMock}>
                    {moviesDataMock.map((movie) => (
                        <Link to={`/movie/${movie.id}`} key = {movie.id} />
                    ))}
                </MoviesList>
            </MemoryRouter>
        );
        expect(component.find(Link).props().to).toBe('/movie/1');
    });
});
