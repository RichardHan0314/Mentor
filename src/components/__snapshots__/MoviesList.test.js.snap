// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component Render NotFindURL snapshot 1`] = `ShallowWrapper {}`;
